cmake/modules/autopas_auto4omp.cmake

```diff
@@ -0,0 +1,576 @@
+#[=====================================================================================================================[
+File: autopas_auto4omp.cmake
+Author: MehdiHachicha
+Date: 15.04.2024
+
+This CMake module loads Auto4OMP into the AutoPas project.
+It attempts to automatically handle its required CMake arguments, and warns if one must be manually entered.
+Ideally, "cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++" should work without issues.
+If cuda's installed, a gcc compatible with NVCC should also be installed. E.g., gcc-12.
```

Comment:
If CUDA's installed on the system, LB4OMP will attempt to link it, then fail the build because it sets an outdated Compute Capability by default. I originally fixed this by passing it a CMake var to specify a newer CC, didn't find a straightforward way to tell LB4 to ignore CUDA at first. Upon further inspection, seems -DLIBOMPTARGET_NVPTX_ENABLE_BCLIB=OFF disables CUDA, will adjust and test the module and make a second request.

Reply:
If -DLIBOMPTARGET_NVPTX_ENABLE_BCLIB=OFF solves this, I suspect the other compiler setups will work (the clang requirement's under CUDA in the LB4OMP docs). Either way, I'll look into the other compilers a bit more.

Reply:
Update: -DOPENMP_ENABLE_LIBOMPTARGET=OFF is the solution. It removes the need for the CUDA portion of the CMake module.